doc/algorithms/quantum_neural_network.rst

```diff
@@ -0,0 +1,178 @@
+
+- Using arbitrary interferometers :math:`U_{k}(\boldsymbol{\theta}_{k},\boldsymbol{\phi}_{k})` with access to phase and general displacement gates (i.e., not necessarily position displacement). This allows :math:`\mathcal{D} \circ \mathcal{U}_{2} \circ \mathcal{S} \circ \mathcal{U}_{1}` to represent a general Gaussian operation.
+- Using arbitrary non-Gaussian gates :math:`\Phi(\lambda_{i})`, such as the Kerr gate.
+- Encoding data outside of the position eigenbasis, for example using instead the Fock basis.
+
+In fact, gates in a single layer form a universal gate set, making the CV quantum neural network a model for universal quantum computing, i.e., a sufficient number of layers can carry out any quantum algorithm implementable on a CV quantum computer.
+
+CV quantum neural networks can be trained both through classical simulation and directly on quantum hardware. Strawberry Fields relies on classical simulation to evaluate cost functions of the CV quantum neural network and the resultant gradients with respect to parameters of each layer. However, this becomes an intractable task with increasing network depth and width. Ultimately, direct evaluation on hardware is the feasible approach for large scale networks, with an approach suggested in :cite:`schuld2019evaluating`. The `PennyLane <https://pennylane.readthedocs.io/en/latest/>`_ library provides tools for training hybrid quantum-classical machine learning models with both simulators and quantum hardware.
+
+CV quantum neural network layers are shown for one to four modes below:
+
+:html:`<br>`
+
+.. figure:: ../_static/layer_1mode.svg
+    :align: center
+    :width: 31%
+    :target: javascript:void(0);
+
+    One mode layer
+
+:html:`<br>`
+
+
+.. figure:: ../_static/layer_2mode.svg
+    :align: center
+    :width: 46%
+    :target: javascript:void(0);
+
+    Two mode layer
+
+:html:`<br>`
+
+
+
+.. figure:: ../_static/layer_3mode.svg
+    :align: center
+    :width: 75%
+    :target: javascript:void(0);
+
+    Three mode layer
+
+:html:`<br>`
+
+.. figure:: ../_static/layer_4mode.svg
+    :align: center
+    :width: 90%
+    :target: javascript:void(0);
+
+    Four mode layer
+
+:html:`<br>`
+
+Here, the multimode linear interferometers :math:`U_{1}` and :math:`U_{2}` have been decomposed into two-mode phaseless beamsplitters (:class:`~.BSgate`) and single-mode phase shifters (:class:`~.Rgate`) using the Clements decomposition :cite:`clements2016`. The Kerr gate is used as the non-Gaussian gate.

```

Comment:
Do we want to refer to other names we use for decompositions? i.e., `rectangular` or `triangular`?

Reply:
I feel that we should do this separately across the code base in a new PR. It would be confusing to use the terms rectangular and triangular here, when we currently don't use them anywhere else.